homeassistant/components/harmony/remote.py

```diff
@@ -65,65 +58,36 @@
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Harmony platform."""
-    activity = None
-
-    if CONF_DEVICE_CACHE not in hass.data:
-        hass.data[CONF_DEVICE_CACHE] = []
 
     if discovery_info:
-        # Find the discovered device in the list of user configurations
-        override = next(
-            (
-                c
-                for c in hass.data[CONF_DEVICE_CACHE]
-                if c.get(CONF_NAME) == discovery_info.get(CONF_NAME)
-            ),
-            None,
+        # Now handled by ssdp in the config flow
+        return
+
+    if CONF_HOST not in config:
+        _LOGGER.error(
+            "The harmony remote '%s' cannot be setup because configuration now requires a host when configured manually.",
```

Comment:
This should be handled by the config schema.

Reply:
@MartinHjelmare Can you give me a little more color on this one?  I'm not sure how to start an import that is user interactive here.

Reply:
If we require the host in the config we should require it in the config schema too. Right now it's optional. So we should change to make it required with `vol.Required`.

Reply:
Looks like I can take out the check now that I made it required.

done